addons/web/static/src/legacy/js/core/cookie_utils.js

```diff
@@ -16,6 +16,8 @@ const utils = {
             var cookie = parts.join('=');
 
             if (cookieName && cookieName === name) {
+                if (cookie.startsWith('"'))
+                    cookie = cookie.slice(1, -1);
```

Comment:
This is a partial solution, the actual algorithm used within werkzeug is much more complexe and involve various escaping mechanism, see the actual quoting/unquoting algorithm: https://github.com/pallets/werkzeug/blob/2b2c4c3dd3cf7389e9f4aa06371b7332257c6289/src/werkzeug/_internal.py#L271-L321

I'm ok we don't code a complete solution *right-now* but I would rather do the following safer solution:

```suggestion
                if (cookie.startsWith('"')) {
                    if (cookie.contains('\\')) {
                        // see werkzeug _cookie_quote
                        throw new Error(`Cookie value contains unknown characters {cookie}`);  
                    }
                    cookie = cookie.slice(1, -1);
                }
                    
```

@odoo/rd-framework-js will know better than I do

Reply:
You are absolutely correct. I've missed that whole method for cookie unquote.
Let me see what can be done there.
Stay tuned :)

Reply:
Hi @Julien00859,

I've applied your suggestion.

Cheers,
Petar